jwksx/fetcher_v2.go

```diff
@@ -156,7 +169,7 @@ func (f *FetcherNext) fetch(ctx context.Context, location string, opts *fetcherN
 
 	set, err := jwk.ParseReader(result)
 	if err != nil {
-		return nil, err
+		return nil, &JwkParseError{msg: "failed to parse JWK set", cause: err}
```

Comment:
Probably makes sense to just return `errors.WithStack(herodot.ErrBadRequest.WithReason("...").WithErr(err))`